modules/photon/photon.js

```diff
@@ -1,54 +1,60 @@
 /* jshint onevar: false */
 
-( function( $ ) {
+( function() {
+	function recalculate() {
+		if ( this.complete ) {
+			// Support for lazy loading: if there is a lazy-src attribute and it's value
+			// is not the same as the current src we should wait until the image load event
+			var lazySrc = this.getAttribute('data-lazy-src');
```

Comment:
I'm not familiar with how Lazy loading attributes work and if they are injected from backend or in frontend with JS, so just leaving some notes in case there are concerns to test:

- `skip-lazy` css class might be present, can it occur with `data-lazy-src` simultaneously? ([docs](https://jetpack.com/support/lazy-images/))
- Can this script run before Lazy scripts and can that affect if `data-lazy-src` is present?

Reply:
> skip-lazy css class might be present, can it occur with data-lazy-src simultaneously?

According to lazy load code - `skip-lazy` is in the blacklisted classed, which bails before `data-lazy-src` is getting set.

> Can this script run before Lazy scripts and can that affect if data-lazy-src is present?

photon scripts have higher priority(9), which makes sure it gets loaded before lazy images. Lazy load script relies on jQuery `ready` function. But photon relies on `complete` to be true, which (I'd expect) to resolve after lazy images scripts would finish its work. FWIW we do not change any functionality in this PR